schema_registry/serializers/message_serializer.py

```diff
@@ -80,10 +80,6 @@ def encode_record_with_schema(self, subject: str, schema: typing.Union[BaseSchem
         # Try to register the schema
         schema_id = self.schemaregistry_client.register(subject, schema, schema_type=self._serializer_schema_type)
 
-        if not schema_id:
-            message = f"Unable to retrieve schema id for subject {subject}"
-            raise SerializerError(message)
-
```

Comment:
@marcosschroh this code is unreachable. `schemaregistry_client.register()` would never return `None`. It'd raise `ClientError` in case there are errors in the registration of the schema. So I removed the code to increase code coverage 😬